llmeter/results.py

```diff
@@ -34,10 +33,31 @@ class Result:
     provider: str | None = None
     run_name: str | None = None
     run_description: str | None = None
+    start_time: float | None = None
+    end_time: float | None = None
 
     def __str__(self):
         return json.dumps(self.stats, indent=4, default=str)
 
+    def __post_init__(self):
+        """Initialize the Result instance."""
+        self._contributed_stats = {}
+
+    def _update_contributed_stats(self, stats: dict[str, Number]):
+        """
+        Upsert externally-provided statistics for the `stats` property
+
+        Callbacks can use this method to extend the default stats with additional key-value pairs.
+        """
+        if not isinstance(stats, dict):
+            raise ValueError("Stats must be a dictionary")
+        for key, value in stats.items():
+            if not isinstance(value, Number):
+                raise ValueError(
+                    f"Value for key {key} must be a number, got {type(value)}"
+                )
```

Comment:
to be able to update/complement arbitrary modifications we would need a comprehensive lineage/undo history. that sounds like overkill. possible alternative approach, cost (and any additional custom record-level entries) should be forced to live in a separate file with referencing records by index. this would allow multiple cost analysis on the same initial data, and they would all coexist.
Thoughts?